migration-engine/migration-engine-tests/tests/create_migration/create_migration_tests.rs

```diff
@@ -321,6 +321,171 @@ fn creating_a_migration_with_a_non_existent_migrations_directory_should_work(api
+fn alter_enum_name_remove_and_change_default(mut api: TestApi) {
+    let plain_dm = api.datamodel_with_provider(
+        r#"
+        model Cat {
+            id      Int  @id
+            mood    Mood @default(HUNGRY)
+        }
+
+        enum Mood {
+            HUNGRY
+            SLEEPY
+        }
+    "#,
+    );
+
+    let dir = api.create_migrations_directory();
+    api.create_migration("plain", &plain_dm, &dir).send_sync();
+
+    let custom_dm = api.datamodel_with_provider(
+        r#"
+        model Cat {
+            id      Int  @id
+            mood    Mood @default(MOODY)
+        }
+
+        enum Mood {
+            SLEEPY
+            MOODY
+        }
+    "#,
+    );
+
+    let is_postgres = api.is_postgres();
+    let is_cockroach = api.is_cockroach();
+
+    api.create_migration("custom", &custom_dm, &dir)
+        .send_sync()
+        .assert_migration_directories_count(2)
+        .assert_migration("custom", move |migration| {
+            let expected_script = if is_cockroach {
+                expect![[r#"
+                    /*
+                      Warnings:
+
+                      - The values [HUNGRY] on the enum `Mood` will be removed. If these variants are still used in the database, this will fail.
+
+                    */
+                    -- AlterEnum
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" DROP DEFAULT;
+                    ALTER TYPE "Mood" ADD VALUE 'MOODY';
+                    ALTER TYPE "Mood" DROP VALUE 'HUNGRY';
+
+                    -- AlterTable
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" SET DEFAULT 'MOODY';
+                "#]]
+            } else if is_postgres {
+                expect![[r#"
+                    /*
+                      Warnings:
+
+                      - The values [HUNGRY] on the enum `Mood` will be removed. If these variants are still used in the database, this will fail.
+
+                    */
+                    -- AlterEnum
+                    BEGIN;
+                    CREATE TYPE "Mood_new" AS ENUM ('SLEEPY', 'MOODY');
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" DROP DEFAULT;
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" TYPE "Mood_new" USING ("mood"::text::"Mood_new");
+                    ALTER TYPE "Mood" RENAME TO "Mood_old";
+                    ALTER TYPE "Mood_new" RENAME TO "Mood";
+                    DROP TYPE "Mood_old";
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" SET DEFAULT 'MOODY';
+                    COMMIT;
+
+                    -- AlterTable
+                    ALTER TABLE "Cat" ALTER COLUMN "mood" SET DEFAULT 'MOODY';
+                "#]]
+            } else {
+                panic!()
+            };
+
+            migration.expect_contents(expected_script)
+        });
+}
+
+#[test_connector(tags(Postgres, CockroachDb))]
+fn alter_enum_name_with_default_empty_list_shouldnt_set_new_default(mut api: TestApi) {
```

Comment:
This test would panic exactly like https://github.com/prisma/prisma/issues/15705 before this PR was introduced.